Java/core/src/main/java/com/amazon/randomcutforest/tree/RandomCutTree.java

```diff
@@ -739,9 +712,7 @@ public float[] getPointSum(int index) {
         checkArgument(centerOfMassEnabled, " enable center of mass");
         if (isLeaf(index)) {
             float[] point = projectToTree(pointStoreView.get(getPointIndex(index)));
-            if (point.length != dimension) {
-                throw new IllegalStateException(" incorrect projection");
-            }
+            checkArgument(point.length != dimension, () -> " incorrect projection");
```

Comment:
Should it be point.length == dimension?

Reply:
yes, changed, the new push should overwrite